system-addon/lib/TopStoriesFeed.jsm

```diff
@@ -311,7 +311,7 @@ this.TopStoriesFeed = class TopStoriesFeed {
         // Create a new array with a spoc inserted at index 2
         const position = SectionsManager.sections.get(SECTION_ID).order;
         let rows = this.store.getState().Sections[position].rows.slice(0, this.stories.length);
-        rows.splice(2, 0, spocs[0]);
+        rows.splice(2, 0, Object.assign(spocs[0], {pinned: true}));
 
         // Send a content update to the target tab
         const action = {type: at.SECTION_UPDATE, data: Object.assign({rows}, {id: SECTION_ID})};
```

Comment:
Might not be an issue now, but potentially this `SECTION_UPDATE` could result in multiple spocs trying to be in index 2 if some reason `maybeAdd` is called multiple times for a given page?

Reply:
That would be the case if NEW_TAB_REHYDRATED is fired multiple times for a tab, yes, but we do have a check to only fire this event once: https://github.com/mozilla/activity-stream/blob/master/system-addon/content-src/components/Base/Base.jsx#L46

Reply:
Oh I was thinking more of in the future if we had multiple callers of `maybeAdd` where we would need to be extra careful. But we're not there now, so maybe not a real issue! ;)

Reply:
OK, I guess let's wait and see if the spoc feature sticks around and how, once experimentation ends. Thanks for your help!